### ISSUE TITLE
[Request]: Autorun application

### ISSUE THREAD

**Issue by zDEFz** [bottlesdevs/Bottles#1932]
### Tell us the problem or your need

I'd like to autostart a program I always need

### Describe the solution you'd like

To run the bottle at startup

**Comment by mirkobrombin** [comment]
Please provide more info

**Comment by zDEFz** [comment]
What info do you need? For instance, if you want to autorun any bottle program, how would you do that? I know how it works in wine commandline, but how do you call a bottle from the commandline and run it?

**Comment by mirkobrombin** [comment]
I'm missing the "autorun" concept. I mean, are you talking about like running apps automatically when opening a bottle?

**Comment by zDEFz** [comment]
> I'm missing the "autorun" concept. I mean, are you talking about like running apps automatically when opening a bottle?

No. I'd like the host (Linux) to start a bottle of my choice automatically on bootup. Not programs in the bottle to start other programs.
Linux Starts program X in bottle X automatically. Application starts appears. Me happy.

**Comment by axtloss** [comment]
A (temporary) solution is to just create a desktop entry with bottles of the application and then move the generated .desktop file in `~/.local/share/applications` to `~/.config/autostart`, you may have to create the directory if it doesn't exist

**Comment by mirkobrombin** [comment]
I think this is the way to go, avoiding using a service. It seems to me a very specific feature, I don't know if it should be implemented or let the users create Desktop entries on their own.

**Comment by zDEFz** [comment]
I might add that the autostart directory depends. Is there a way to detect which autostart directory is currently the right one and give an option to save the .desktop file into that folder? might require temporary sudo access.

**Comment by axtloss** [comment]
Every de should follow the ~/.config/autostart directory since I'm pretty sure that it's a desktop standard, the only extra permission bottles would need for that is read-write access to that directory if it's a flatpak.
Additionally you could just use the tools of the de (gnome-tweaks if it's gnome) to add the application to autostart, assuming you've created the desktop entry.

**Comment by zDEFz** [comment]
> Every de should follow the ~/.config/autostart directory since I'm pretty sure that it's a desktop standard, the only extra permission bottles would need for that is read-write access to that directory if it's a flatpak. Additionally you could just use the tools of the de (gnome-tweaks if it's gnome) to add the application to autostart, assuming you've created the desktop entry.

it depends. With xdg, its /etc/xdg/autostart

**Comment by axtloss** [comment]
/etc/xdg/autostart is for system wide stuff, ~/.config/autostart is only for the user.

/usr/share/applications are desktop entries that are system wide, while ~/.local/share/applications is only for the user

**Comment by mirkobrombin** [comment]
I prefer to document the process in our documentation and avoid more holes in the flatpak
